### PR TITLE
PHATE v1.0.0

### PR DESCRIPTION
**Pull request by scottgigante**
Changeset:
* added `optimizer='sgd'`
* clean up tutorials
* added DOI reference for Nature Biotech publication

### CODE REVIEW

**Comment by scottgigante** [comment]
@dburkhardt thoughts on the optimizer API? https://github.com/KrishnaswamyLab/PHATE/pull/74/files#diff-9196f1cb9654b39330de4de7d9c51356R77

File: Python/phate/phate.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ class PHATE(BaseEstimator):
         n_pca < 20 allows neighborhoods to be calculated in
         roughly log(n_samples) time.
 
+    optimizer : {'sgd', 'smacof'}, optional (default: 'sgd')
```
Comment (dburkhardt) [inline Python/phate/phate.py]: I'm not sure the policy of default prioritizing speed over optimality is better here. I would feel much better if the optimal solver was default.

File: Python/phate/phate.py [hunk 2]
```diff
@@ -443,98 +520,102 @@ def set_params(self, **params):
...
-            self.n_jobs = params['n_jobs']
-            self._set_graph_params(n_jobs=params['n_jobs'])
-            del params['n_jobs']
-        if 'random_state' in params:
-            self.random_state = params['random_state']
-            self._set_graph_params(random_state=params['random_state'])
-            del params['random_state']
-        if 'verbose' in params:
-            self.verbose = params['verbose']
+        if "n_jobs" in params:
```
Comment (dburkhardt) [inline Python/phate/phate.py]: Aww, I liked single ticks

File: Python/phate/phate.py (hunk 1 above)
Comment (dburkhardt) [reply Python/phate/phate.py]: Also for the API, TSNE in sklearn uses `method=` to specify the barnes-hut vs exact optimization (https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE)

>_New in version 0.17:_ Approximate optimization _method via the Barnes-Hut._ [emphasis theirs]

Isomap uses `eigen_solver=` to specify the eigenvalue decomposition method.

What about `mds_solver=` for PHATE?

File: Python/phate/phate.py (hunk 2 above)
Comment (scottgigante) [reply Python/phate/phate.py]: `black` says otherwise, fight them not me

File: Python/phate/phate.py (hunk 1 above)
Comment (scottgigante) [reply Python/phate/phate.py]: I like mds_solver, good idea

Comment (scottgigante) [reply Python/phate/phate.py]: I like `mds_solver`, good idea.

File: Python/phate/phate.py (hunk 2 above)
Comment (scottgigante) [reply Python/phate/phate.py]: Fight `black` not me

**Comment by scottgigante** [comment]
@dburkhardt can I go ahead and merge?

**Pull request merged**